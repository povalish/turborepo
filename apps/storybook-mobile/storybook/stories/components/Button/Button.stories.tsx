import React from 'react';
import { Text } from 'react-native';

import { action } from '@storybook/addon-actions';
import { text } from '@storybook/addon-knobs';
import { storiesOf } from '@storybook/react-native';

import { Button } from 'ui-kit/components/Button/Button.native';
import { Wrapper } from '../../layouts/Wrapper';

// Button.stories
//

storiesOf('Button', module)
  .addDecorator((getStory) => <Wrapper>{getStory()}</Wrapper>)
  .add('Textable', () => (
    <Button onPress={action('clicked-text')}>
      <Text>{text('Button text', 'Hello Button')}</Text>
    </Button>
  ))
  .add('Emoji', () => (
    <Button onPress={action('clicked-emoji')}>
      <Text>😀 😎 👍 💯</Text>
    </Button>
  ));
