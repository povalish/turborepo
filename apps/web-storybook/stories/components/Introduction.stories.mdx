<Meta title="Components/Introduction" />

# UI Components

Storybook for Web UI Library
