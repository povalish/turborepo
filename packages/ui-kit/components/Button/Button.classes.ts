import { cva } from 'class-variance-authority';

// Button.classes.ts
//

export const buttonClasses = cva(`
  min-w-[160px] py-1.5 px-2 rounded-lg
  bg-[#FCB07E] border-[#FCB07E] border-2
`);

export const textClasses = cva(`
  text-[#333E53] text-xl text-center
  font-bold
`);

export const classes = cva(`${buttonClasses()} ${textClasses()}`);
