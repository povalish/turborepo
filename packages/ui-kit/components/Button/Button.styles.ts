import { cva } from 'class-variance-authority';

export const styles = cva(`
  min-w-[160px] py-1.5 px-2 rounded-lg
  bg-[#FCB07E] border-[#FCB07E] border-2
  text-[#333E53] font-bold text-xl
`);
