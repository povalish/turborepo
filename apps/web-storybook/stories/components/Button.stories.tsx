import React from 'react';
import { ComponentStory, ComponentMeta } from '@storybook/react';
import { Button } from 'ui-kit/components/Button';

// Button.stories
//

export default {
  title: 'Components/Button',
  component: Button,
} as ComponentMeta<typeof Button>;

const Template: ComponentStory<typeof Button> = (args) => <Button {...args}>Simple Button</Button>;

export const PrimaryButton = Template.bind({});
