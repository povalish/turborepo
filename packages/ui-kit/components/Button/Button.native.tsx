import React, { ComponentProps, PropsWithChildren } from 'react';
import { styled } from 'nativewind';
import { TouchableOpacity, Text } from 'react-native';
import { buttonClasses, textClasses } from './Button.classes';

export const ButtonView = styled(TouchableOpacity, buttonClasses());
export const ButtonText = styled(Text, textClasses());

// Button.native
//

interface IButton extends PropsWithChildren, ComponentProps<typeof ButtonView> {
  textClassName?: string;
}

export const Button: React.FC<IButton> = ({ children, className, textClassName, ...props }) => (
  <ButtonView activeOpacity={0.8} className={buttonClasses({ className })} {...props}>
    <ButtonText className={textClasses({ className: textClassName })}>{children}</ButtonText>
  </ButtonView>
);
