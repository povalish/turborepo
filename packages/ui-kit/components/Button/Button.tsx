import React, { ButtonHTMLAttributes } from 'react';
import { styles } from './Button.styles';

// Button
//

interface IButton extends ButtonHTMLAttributes<HTMLButtonElement> {}

export const Button: React.FC<IButton> = ({ children, className, ...buttonProps }) => (
  // Button type has come from props
  // eslint-disable-next-line react/button-has-type
  <button className={styles({ className })} {...buttonProps}>
    {children}
  </button>
);
