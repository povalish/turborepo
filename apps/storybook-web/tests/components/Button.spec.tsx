import { render } from '@testing-library/react';
import { Button } from 'ui-kit/components/Button';

describe('ui components > Button', () => {
  it('renders component unchanged', () => {
    const { container } = render(<Button />);
    expect(container).toMatchSnapshot();
  });
});
