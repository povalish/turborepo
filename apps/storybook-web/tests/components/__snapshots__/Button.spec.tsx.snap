// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ui components > Button renders component unchanged 1`] = `
<div>
  <button
    class="
  min-w-[160px] py-1.5 px-2 rounded-lg
  bg-[#FCB07E] border-[#FCB07E] border-2
 
  text-[#333E53] text-xl text-center
  font-bold
"
  />
</div>
`;
